### PR TITLE
feat(core): Refactor ImageGeneration traits and introduce HttpClient

### PR DESCRIPTION
**Pull request by mandeepsingh2007**
##Issue
#497 

## Summary
This PR is the first part of the Image Generation refactoring series (1/5). It establishes the core traits and infrastructure needed for supporting new providers like Vertex AI and Bedrock, without breaking existing functionality.

## Changes
- **Core Traits**: Updated `ImageGenerationClient` and `ImageGenerationConfig` to support more flexible configuration and future provider integrations.
- **HttpClient**: Introduced a `HttpClient` wrapper to abstract HTTP operations, improving testability and allowing for future backend swaps.
- **Refactoring**: Updated existing `OpenAIImageClient`, `StableDiffusionClient`, and `HuggingFaceClient` to use the new `HttpClient` and traits.
- **Cleanup**: Removed deprecated code and improved type safety.

## Note to Reviewers
This PR purposely excludes new providers (Vertex AI, Bedrock, etc.) and metrics instrumentation to keep the review manageable. These will follow in subsequent PRs stacking on top of this one.

### CODE REVIEW

**Comment by Sukuna0007Abhi** [comment]
Add some  more edge cases tests for this

File: format_all_prs.sh [hunk 1]
```diff
@@ -0,0 +1,50 @@
+
+# PR 1: Core
+git checkout image-gen-pr1-core
```
Comment (vim89) [inline format_all_prs.sh]: Nit but important. This repository level branch management script is not part of library runtime behavior and does not belong in this refactor PR. Guideline principle is single responsibility and minimal change surface per PR. Please move this to local tooling docs or drop it from this PR.

File: modules/core/src/main/scala/org/llm4s/imagegeneration/ImageGeneration.scala [hunk 2]
```diff
@@ -210,24 +343,66 @@ trait ImageGenerationClient {
...
-        val httpClient = HttpClient.createHttpClient(hfConfig)
+        val httpClient = HttpClient.create()
         new HuggingFaceClient(hfConfig, httpClient)
       case openAIConfig: OpenAIConfig =>
-        new OpenAIImageClient(openAIConfig)
+        val httpClient = HttpClient.create()
+        new OpenAIImageClient(openAIConfig, httpClient)
+      case _ =>
+        // For PR flow, other providers (Vertex, Bedrock, etc.) are not yet available in this branch
+        throw new UnsupportedOperationException(s"Provider ${config.provider} is not yet integrated in this version.")
```
Comment (vim89) [inline modules/core/src/main/scala/org/llm4s/imagegeneration/ImageGeneration.scala]: Request changes. This introduces a runtime throw in core factory flow for providers that are already part of public config and provider ADT. Guideline principle is no throw in domain logic and fail with typed errors. Bigger picture this weakens reliability and makes extension feel partially wired but unusable at runtime.

File: modules/core/src/main/scala/org/llm4s/imagegeneration/provider/OpenAIImageClient.scala [hunk 3]
```diff
@@ -79,61 +80,164 @@ class OpenAIImageClient(config: OpenAIConfig) extends ImageGenerationClient {
...
+            case Right(_) =>
+              Left(UnknownError(new RuntimeException("Unexpected successful response during error handling")))
+          }
+        }
+      }
+    }
+
+  /**
+   * Generate an image asynchronously
+   */
```
Comment (vim89) [inline modules/core/src/main/scala/org/llm4s/imagegeneration/provider/OpenAIImageClient.scala:150]: Should fix. Async methods are wrapping blocking HTTP path inside Future on caller execution context. Guideline principle is avoid blocking on global execution context and keep effect boundaries explicit. Please use a dedicated blocking dispatcher or keep async API out until non blocking path is added.

File: modules/core/src/main/scala/org/llm4s/imagegeneration/ImageGeneration.scala [hunk 4]
```diff
@@ -138,10 +168,15 @@ object ImageGenerationProvider {
   case object DALLE           extends ImageGenerationProvider
   case object Midjourney      extends ImageGenerationProvider
   case object HuggingFace     extends ImageGenerationProvider
+  case object VertexAI        extends ImageGenerationProvider
```
Comment (vim89) [inline modules/core/src/main/scala/org/llm4s/imagegeneration/ImageGeneration.scala]: Should fix. Adding new providers and config types in core before integration creates a public API that fails at runtime in factory dispatch. Guideline principle is fail at compile time where possible and avoid extension points that are not operational. Either ship end to end per provider or keep these types out of this PR.

File: modules/core/src/main/scala/org/llm4s/imagegeneration/ImageGeneration.scala [hunk 5]
```diff
@@ -275,14 +487,14 @@ object ImageGeneration {
...
+   * Get an OpenAI client with the required API key.
    *
    * This is a convenience method for creating a client that connects to the
-   * OpenAI API for DALL-E image generation.
+   * OpenAI API for image generation.
    *
    * @param apiKey Your OpenAI API key (required).
-   * @param model The DALL-E model version to use. Defaults to dall-e-2.
-   * @return An `ImageGenerationClient` instance configured for OpenAI DALL-E.
+   * @param model The model version to use. Defaults to gpt-image-1.
```
Comment (vim89) [inline modules/core/src/main/scala/org/llm4s/imagegeneration/ImageGeneration.scala:409]: Must fix. The doc now says default is gpt-image-1, but openAIClient still defaults to dall-e-2 in the signature below. Guideline principle is API contract and implementation must stay consistent. Please align the method default and docs in the same change.

**Comment by mandeepsingh2007** [comment]
@vim89 @rorygraves sir please review this pr i have fixed the changes you told

**Comment by mandeepsingh2007** [comment]
@rorygraves sir please review this pr i have fixed all the changes
